fenix/app/src/main/java/org/mozilla/fenix/compose/TabCounter.kt

```diff
@@ -0,0 +1,120 @@
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.semantics.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.tabstray.TabsTrayTestTag
+
+private const val MAX_VISIBLE_TABS = 99
+private const val SO_MANY_TABS_OPEN = "∞"
+private val NORMAL_TABS_BOTTOM_PADDING = 0.5.dp
+private const val ONE_DIGIT_SIZE_RATIO = 0.5f
+private const val TWO_DIGITS_SIZE_RATIO = 0.4f
+
+/**
+ * UI for displaying the number of opened tabs.
+ *
+ * @param tabCount the number to be displayed inside the counter.
+ */
+
+@Composable
+@Suppress("MagicNumber")
+fun TabCounter(tabCount: Int) {
+    val normalTabCountText: String
+    val tabCountTextRatio: Float
+    val needsBottomPaddingForInfiniteTabs: Boolean
+
+    when (tabCount) {
+        in 0..9 -> {
+            normalTabCountText = tabCount.toString()
+            tabCountTextRatio = ONE_DIGIT_SIZE_RATIO
+            needsBottomPaddingForInfiniteTabs = false
+        }
+
+        in 10..MAX_VISIBLE_TABS -> {
+            normalTabCountText = tabCount.toString()
+            tabCountTextRatio = TWO_DIGITS_SIZE_RATIO
+            needsBottomPaddingForInfiniteTabs = false
+        }
+
+        else -> {
+            normalTabCountText = SO_MANY_TABS_OPEN
+            tabCountTextRatio = ONE_DIGIT_SIZE_RATIO
+            needsBottomPaddingForInfiniteTabs = true
+        }
+    }
+
+    val normalTabsContentDescription = if (tabCount == 1) {
+        stringResource(id = R.string.mozac_tab_counter_open_tab_tray_single)
+    } else {
+        stringResource(
+            id = R.string.mozac_tab_counter_open_tab_tray_plural,
+            tabCount.toString(),
+        )
+    }
+
+    val counterBoxWidthDp =
+        dimensionResource(id = mozilla.components.ui.tabcounter.R.dimen.mozac_tab_counter_box_width_height)
+    val counterBoxWidthPx = LocalDensity.current.run { counterBoxWidthDp.roundToPx() }
+    val counterTabsTextSize = (tabCountTextRatio * counterBoxWidthPx).toInt()
+
+    val normalTabsTextModifier = if (needsBottomPaddingForInfiniteTabs) {
+        val bottomPadding = with(LocalDensity.current) { counterTabsTextSize.toDp() / 4 }
+        Modifier.padding(bottom = bottomPadding)
+    } else {
+        Modifier.padding(bottom = NORMAL_TABS_BOTTOM_PADDING)
+    }
+
+    Box(
+        modifier = Modifier
+            .semantics(mergeDescendants = true) {
+                testTag = TabsTrayTestTag.normalTabsCounter
+            },
+        contentAlignment = Alignment.Center,
+    ) {
+        Icon(
+            painter = painterResource(
+                id = mozilla.components.ui.tabcounter.R.drawable.mozac_ui_tabcounter_box,
+            ),
+            contentDescription = normalTabsContentDescription,
+//            tint = FirefoxTheme.colors.iconPrimary,
```

Comment:
Why is this commented out and then in `NavigationBar` we use a `CompositionLocalProvider` for the same color?